src/main/java/cloud/fogbow/common/util/connectivity/cloud/CloudHttpClient.java

```diff
@@ -32,28 +38,55 @@ public String doPostRequest(String url, String bodyContent, T cloudUser)
         return callDoGenericRequest(HttpMethod.POST, url, bodyContent, cloudUser);
     }
 
-    private String callDoGenericRequest(HttpMethod method, String url, String bodyContent, T cloudUser)
+    @VisibleForTesting
+    String callDoGenericRequest(HttpMethod method, String url, String bodyContent, T cloudUser)
             throws FogbowException, HttpResponseException {
-        HashMap<String, String> body = GsonHolder.getInstance().fromJson(bodyContent, HashMap.class);
 
         HashMap<String, String> headers = new HashMap<>();
+        HashMap<String, String> body = GsonHolder.getInstance().fromJson(bodyContent, HashMap.class);
         HttpResponse response = doGenericRequest(method, url, headers, body, cloudUser);
-
         if (response.getHttpCode() > HttpStatus.SC_NO_CONTENT) {
-            throw new HttpResponseException(response.getHttpCode(), response.getContent());
+            if (response.getHttpCode() == SC_REQUEST_HEADER_FIELDS_TOO_LARGE) {
+                // When the status code is 431, the error message must be obtained in the
+                // response headers.
+                Map<String, List<String>> responseHeaders = response.getHeaders();
+                throw new HttpResponseException(response.getHttpCode(), getMessageFrom(responseHeaders));
+            } else {
+                throw new HttpResponseException(response.getHttpCode(), response.getContent());
+            }
         }
-
         return response.getContent();
     }
 
+    @VisibleForTesting
+    String getMessageFrom(Map<String, List<String>> responseHeaders) {
+        String message = null;
+        if (responseHeaders.containsKey(CloudStackConstants.X_DESCRIPTION_KEY)) {
```

Comment:
This "X_Description" key is only used by Cloudstack, isn't it ? And this class is used by Openstack and Cloudstack, rigth? Thus, I think this logic is in the wrong place. Lets discuss it.

Reply:
I believe that we should think about a later refactoring solution for this.